components/builder-originsrv/src/migrations/origin_packages.rs

```diff
@@ -267,5 +317,32 @@ pub fn migrate(migrator: &mut Migrator) -> Result<()> {
                         RETURN;
                     END
                     $$ LANGUAGE plpgsql STABLE"#)?;
+    migrator.migrate("originsrv",
+                     r#"CREATE OR REPLACE FUNCTION get_origin_packages_for_version_v1 (
+                            op_ident text,
+                            op_limit bigint,
+                            op_offset bigint
+                        ) RETURNS TABLE(total_count bigint, ident text, channels text[]) AS
+                        $$
+                            BEGIN
+                                RETURN QUERY
+                                   SELECT COUNT(*) OVER ()
+                                       AS total_count,
+                                          origin_packages.ident,
+                                          (SELECT ARRAY(SELECT name
+                                                          FROM origin_channels
+                                                         WHERE id IN (SELECT channel_id
+                                                                        FROM origin_channel_packages
+                                                                       WHERE package_id = origin_packages.id
+                                                                       ORDER BY created_at DESC)))
+                                     FROM origin_packages
+                                    WHERE origin_packages.ident LIKE (op_ident  || '%')
+                                 ORDER BY ident DESC
+                                    LIMIT op_limit
+                                   OFFSET op_offset;
+                                RETURN;
```

Comment:
Also, just to confirm the intent: this query returns 
a) the total number of packages that satisfy the query string
b) the identifiers of those packages that fall within our `LIMIT` / `OFFSET` window
and c) all the channels that those individual packages are currently in

Is that accurate?